src/execution_plan/ops/op_index_scan.c

```diff
@@ -56,73 +78,150 @@ static inline void _UpdateRecord(IndexScan *op, Record r, EntityID node_id) {
 	Node n = GE_NEW_LABELED_NODE(op->n.label, op->n.label_id);
 	int res = Graph_GetNode(op->g, node_id, &n);
 	ASSERT(res != 0);
-	// Get a pointer to the node's allocated space within the Record.
 	Record_AddNode(r, op->nodeRecIdx, n);
 }
 
-static Record IndexScanConsumeFromChild(OpBase *opBase) {
-	IndexScan *op = (IndexScan *)opBase;
+static inline bool _PassUnresolvedFilters(const IndexScan *op, Record r) {
+	bool           res                  =  true;
+	FT_FilterNode  *unresolved_filters  =  op->unresolved_filters;
```

Comment:
I feel like this function could be more succinctly written as:
```
if(unresolved_filters == NULL) return true;
return (FilterTree_applyFilters(unresolved_filters, r) == FILTER_PASS);
```

Reply:
indeed